src/App.js

```diff
@@ -2,77 +2,50 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  _isMounted = false;
-
   state = {
-    source: 'https://i.imgur.com/VYUdUNJ.png',
+    source: '',
     error: null,
+    loading: true,
   };
 
-  fetchCat() {
-    fetch("https://aws.random.cat/meow").then(res=>res.json()).then(
-      (result) => {
-        this.setState({
-          source: result.file
-        });
-    },
-    (error) => {
-      this.setState({
-        error
-      });
-    }
-    )
-  }
-
   componentDidMount() {
-    this._isMounted = true;
-    const { source } = this.state;
-    this.fetchCat(source);
-
+    this.fetchCat();
   }
 
-  componentWillUnmount() {
-    this._isMounted = false;
+  fetchCat = async () => {
+    try {
+      const result = await fetch("https://aws.random.cat/meow");
+      const image = result.json();
+      this.setState({ source: image.file, loading: false });
+    } catch (e) {
+      this.setState({ error: e.message, loading: false });
+    }
   }
 
   render() {
-    const { error, source } = this.state;
-    if (error) {
-      return <div>Error: {error.message}</div>;
-    }
-    return (
+    const { error, loading, source } = this.state;
+    return loading ? (
+      <p>Loading...</p>
+    ) : error ? (
+      <div>Error: {error}</div>
+    ) : (
       <div className="App">
         <div className="container">
           <div className="row">
             <div className="col-1"></div>
             <div className="col-10">
-              <Image 
-                src={source}
-                className="img-fluid"
-                alt="cat picture"
-              />
+              <img src={source} className="img-fluid" alt="cat picture" />
             </div>
```

Comment:
I removed the two extra components in the bottom because they were serving no purpose, they were an unnecessary abstraction. You had components for the sake of having components. If they needed to be used elsewhere with similar styling or something, then maybe it would justify separating them, but for a one time use, don't abstract or make overly complicated until you NEED to.